lib/connect.ts

```diff
@@ -1,49 +1,74 @@
 import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/distinctUntilChanged';
 import 'rxjs/add/operator/map';
-import { Router, LocationChange } from '@ngrx/router';
-import { Action, Dispatcher, Store } from '@ngrx/store';
+import 'rxjs/add/operator/withLatestFrom';
+import '@ngrx/core/add/operator/select';
+import { Router, Event, NavigationEnd } from '@angular/router';
+import { Location } from '@angular/common';
+import { Store, Action } from '@ngrx/store';
 import { Observable } from 'rxjs/Observable';
 import { Observer } from 'rxjs/Observer';
 
-import { RouterState } from './reducer';
 import { RouterMethodCall, routerActions, routerActionTypes } from './actions';
 
-export function projectLocationChanges(state$: Observable<{ router: RouterState }>): Observable<LocationChange> {
-  return state$.map(s => s.router).filter(change => change !== null);
-}
-
-export function listenForRouterMethodActions(router: Router, actions$: Observable<Action>) {
+export function listenForRouterMethodActions(router: Router, location: Location, actions$: Observable<Action>) {
   actions$
     .filter(action => routerActionTypes.indexOf(action.type) > -1)
     .subscribe(action => {
-      const { path, query }: RouterMethodCall = action.payload;
+      const { path, query: queryParams }: RouterMethodCall = action.payload;
+      let commands: any[] = [path];
 
       switch (action.type) {
         case routerActions.GO:
-          router.go(path, query);
+          router.navigate(commands, { queryParams });
           break;
 
         case routerActions.REPLACE:
-          router.replace(path, query);
+          router.navigate(commands, { queryParams });
```

Comment:
We will also want to add an action for navigating without location change. That wasn't really supported in @ngrx/router.